### ISSUE TITLE
Crash on startForeground() with null notification

### ISSUE THREAD

**Issue by LilPickle3000** [googlecast/CastCompanionLibrary-android#190]
I encountered a crash in the VideoCastNotificationService during poor connectivity.

```
07-16 14:38:07.322: D/ccl_VideoCastNotificat(16539): Build version is: 19
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): [v2.1.1] Failed to set notification for com.google.android.gms.cast.MediaInfo@73f7d8e1
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): com.google.android.libraries.cast.companionlibrary.cast.exceptions.TransientNetworkDisconnectionException
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at com.google.android.libraries.cast.companionlibrary.cast.BaseCastManager.checkConnectivity(BaseCastManager.java:1053)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at com.google.android.libraries.cast.companionlibrary.cast.VideoCastManager.getRemoteMediaInformation(VideoCastManager.java:623)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at com.google.android.libraries.cast.companionlibrary.notification.VideoCastNotificationService.build(VideoCastNotificationService.java:287)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at com.google.android.libraries.cast.companionlibrary.notification.VideoCastNotificationService.access$800(VideoCastNotificationService.java:63)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at com.google.android.libraries.cast.companionlibrary.notification.VideoCastNotificationService$2.onPostExecute(VideoCastNotificationService.java:194)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at com.google.android.libraries.cast.companionlibrary.notification.VideoCastNotificationService$2.onPostExecute(VideoCastNotificationService.java:188)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at android.os.AsyncTask.finish(AsyncTask.java:632)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at android.os.AsyncTask.access$600(AsyncTask.java:177)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:645)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at android.os.Handler.dispatchMessage(Handler.java:102)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at android.os.Looper.loop(Looper.java:146)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at android.app.ActivityThread.main(ActivityThread.java:5487)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at java.lang.reflect.Method.invokeNative(Native Method)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at java.lang.reflect.Method.invoke(Method.java:515)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1283)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1099)
07-16 14:38:07.322: E/ccl_VideoCastNotificat(16539): 	at dalvik.system.NativeStart.main(Native Method)
07-16 14:38:07.342: A/ActivityManager(471): Activity Manager Crash
07-16 14:38:07.342: A/ActivityManager(471): java.lang.IllegalArgumentException: null notification
07-16 14:38:07.342: A/ActivityManager(471): 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:578)
07-16 14:38:07.342: A/ActivityManager(471): 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:14990)
07-16 14:38:07.342: A/ActivityManager(471): 	at android.app.ActivityManagerNative.onTransact(ActivityManagerNative.java:896)
07-16 14:38:07.342: A/ActivityManager(471): 	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2638)
07-16 14:38:07.342: A/ActivityManager(471): 	at android.os.Binder.execTransact(Binder.java:404)
07-16 14:38:07.342: A/ActivityManager(471): 	at dalvik.system.NativeStart.run(Native Method)
```

The cause looks to be a call to startForeground() with a null mNotification. Could be fixed by null-checking before startForeground() call (line 200) in FetchBitmapTask.onPostExecute()  in the same way other calls to startForeground do in VideoCastNotificationService.
```
if (mVisible && (mNotification != null)) {
    startForeground(NOTIFICATION_ID, mNotification);
}
```

**Comment by naddaf** [comment]
Looks like it is actually a bug in the framework, ActiveServices.setServiceForgroundLocker() should exit if notification is null but I guess they don't do a check (there is no contract to not pass a null). Anyway, I will add a check to avoid that in the next update, thanks for reporting this.

**Comment by naddaf** [comment]
Should be fixed in 2.2.
